pkg/reconciler/build/resources/pod.go

```diff
@@ -329,8 +329,7 @@ func MakePod(build *v1alpha1.Build, kubeclient kubernetes.Interface) (*corev1.Po
 			// We execute the build's pod in the same namespace as where the build was
 			// created so that it can access colocated resources.
 			Namespace: build.Namespace,
-			// Ensure our Pod gets a unique name.
-			GenerateName: fmt.Sprintf("%s-", build.Name),
+			Name:      fmt.Sprintf("pod-for-%s", build.Name),
```

Comment:
Good call, done.

FWIW, the reason I dropped `GenerateName` is because the fake k8s implementation doesn't generate names, and so pods were always created as `""`. 😠